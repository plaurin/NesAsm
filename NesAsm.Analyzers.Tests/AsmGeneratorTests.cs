using ApprovalTests.Reporters;

namespace NesAsm.Analyzers.Tests;

[UseReporter(typeof(VisualStudioReporter))]
public class AsmGeneratorTests
{
    [Fact]
    public void TestExampleScript()
    {
        var generator = new AsmGenerator();
        var content = File.ReadAllText($@"..\..\..\..\NesAsm.Example\HiloWorld.cs");

        GeneratorTestUtilities.TestGenerator(generator, content);
    }

    [Fact]
    public void TestSimpleScript()
    {
        TestScript("SimpleScript.cs");
    }

    [Fact]
    public void TestMultiProcScript()
    {
        TestScript("MultiProcScript.cs");
    }

    [Fact]
    public void TestJumpSubroutineScript()
    {
        TestScript("JumpSubroutineScript.cs");
    }

    [Fact]
    public void TestCharsScript()
    {
        TestScript("CharsScript.cs");
    }

    [Fact]
    public void TestBranchScript()
    {
        TestScript("BranchScript.cs");
    }

    [Fact]
    public void TestParametersScript()
    {
        TestScript("ParametersScript.cs");
    }

    [Fact]
    public void TestSubroutineResultScript()
    {
        TestScript("SubroutineResultScript.cs");
    }

    [Fact]
    public void TestSubroutineResultsScript()
    {
        TestScript("SubroutineResultsScript.cs");
    }

    [Fact]
    public void TestCallingOtherScript()
    {
        TestScript("CallingOtherScript.cs");
    }

    [Fact]
    public void TestDataScript()
    {
        TestScript("DataScript.cs");
    }

    [Fact]
    public void TestAllInstructionsScript()
    {
        TestScript("AllInstructionsScript.cs");
    }

    private static void TestScript(string filename)
    {
        var generator = new AsmGenerator();
        var content = File.ReadAllText($@"..\..\..\TestFiles\{filename}");

        GeneratorTestUtilities.TestGenerator(generator, content);
    }
}
