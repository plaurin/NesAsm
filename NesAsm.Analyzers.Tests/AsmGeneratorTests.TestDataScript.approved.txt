- Main diagnostics:
(30,5): error NA0007: Only one of RomDataAttribute or CharDataAttribute can be used at the same time
(34,5): error NA0006: Missing RomDataAttribute or CharDataAttribute on byte[] field
- SyntaxTrees Count: 2
- Output diagnostics:
- RunResult.GeneratedTrees Count: 1
- RunResults diagnostics:
(30,5): error NA0007: Only one of RomDataAttribute or CharDataAttribute can be used at the same time
(34,5): error NA0006: Missing RomDataAttribute or CharDataAttribute on byte[] field
- RunResult[0].Generator: Microsoft.CodeAnalysis.IncrementalGeneratorWrapper
- RunResult[0] diagnostics:
(30,5): error NA0007: Only one of RomDataAttribute or CharDataAttribute can be used at the same time
(34,5): error NA0006: Missing RomDataAttribute or CharDataAttribute on byte[] field
- RunResult[0].GeneratedSources Count: 1
- RunResult[0].Exception: 
- RunResult[0].GeneratedSources[0].HintName: Asm.DataScript.cs
- RunResult[0].GeneratedSources[0].SourceText:
/* Generator Asm code in file Asm.DataScript.s
.segment "CODE"

.proc main

  rts
.endproc

.segment "CODE"

@palettes:
  .byte %00001111
  .byte %00100000
  .byte %00100001
  .byte %00100010
  .byte %00001111
  .byte %00000000
  .byte %00000000
  .byte %00000000

  .byte %00001111
  .byte %00000000
  .byte %00000000
  .byte %00000000
  .byte %00001111
  .byte %00000000
  .byte %00000000
  .byte %00000000

  .byte %00001111
  .byte %00100000
  .byte %00100111
  .byte %00110001
  .byte %00001111
  .byte %00000000
  .byte %00000000
  .byte %00000000

  .byte %00001111
  .byte %00000000
  .byte %00000000
  .byte %00000000
  .byte %00001111
  .byte %00000000
  .byte %00000000
  .byte %00000000


*/
